src/ui/dialogs/spring_pane.rb

```diff
@@ -28,6 +28,8 @@ def initialize
 
     # node_id => period
     @user_periods = {}
```

Comment:
Probably we should put the next three hash into a single one, also called `user_stats` or something...